packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -74,13 +120,34 @@ export class EventSourceMapping extends cdk.Resource {
       throw new Error(`maxBatchingWindow cannot be over 300 seconds, got ${props.maxBatchingWindow.toSeconds()}`);
     }
 
+    if (props.maximumRecordAge && (props.maximumRecordAge.toSeconds() < 60 || props.maximumRecordAge.toSeconds() > 604800)) {
```

Comment:
You could instead use `.toDays()` and check `> 7`. This would be more readable.

Reply:
Had to include {integral: false} when switching to toDays but i still think it's an improvement